### PR TITLE
Add aws_logging collector module, aws preflight and aws_logging_collect recipe

### PR DESCRIPTION
**Pull request by jonathan-greig**

### CODE REVIEW

File: dftimewolf/lib/containers/containers.py [hunk 1]
```diff
@@ -104,6 +104,36 @@ def __init__(self, path, filter_expression, project_name):
...
 
+class AWSLogs(interface.AttributeContainer):
+  """AWS logs container.
+
+  Attributes:
+      path (str): path to a AWS log file.
+      profile_name (str): the profile used to collect logs.
+      query_filter (str): the query filter used in the log query.
+      start_time (str): the start time used in the log query.
+      end_time (str): the end time used in the log query.
```
Comment (tomchop) [inline dftimewolf/lib/containers/containers.py:115]: Maybe just specify the format here too?

File: dftimewolf/lib/containers/containers.py [hunk 2]
```diff
@@ -104,6 +104,36 @@ def __init__(self, path, filter_expression, project_name):
...
+  """
+  CONTAINER_TYPE = 'gcp_logs'
```
Comment (tomchop) [inline dftimewolf/lib/containers/containers.py:117]:
```suggestion
  CONTAINER_TYPE = 'aws_logs'
```

File: dftimewolf/lib/preflights/cloud_token.py [hunk 3]
```diff
@@ -37,4 +40,38 @@ def CleanUp(self):
...
+  def SetUp(self, profile_name=None):  # pylint: disable=arguments-differ
+    """Runs a boto3 check to ensure AWS authentication is configured
+
+    Args:
+      profile_name (str): Optional. The profile to test.
+    """
+
+    try:
+      session = boto_session.Session(profile_name=profile_name)
+      client = session.client('sts')
```
Comment (tomchop) [inline dftimewolf/lib/preflights/cloud_token.py:55]: What does `sts` stand for here? Could you clarify that in a comment?

File: dftimewolf/lib/preflights/cloud_token.py [hunk 4]
```diff
@@ -37,4 +40,38 @@ def CleanUp(self):
...
 
-modules_manager.ModulesManager.RegisterModule(GCPTokenCheck)
+class AWSAccountCheck(module.PreflightModule):
+  """Checks for AWS authentication."""
+
+  def SetUp(self, profile_name=None):  # pylint: disable=arguments-differ
+    """Runs a boto3 check to ensure AWS authentication is configured
+
+    Args:
+      profile_name (str): Optional. The profile to test.
```
Comment (tomchop) [inline dftimewolf/lib/preflights/cloud_token.py:50]: What happens if I don't specify the profile? What gets tested?

File: tests/lib/collectors/aws_logging.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+"""Tests the AWS logging collector."""
+
+import unittest
+
+from dftimewolf.lib import state
+from dftimewolf.lib.collectors import aws_logging
+
+from dftimewolf import config
+
+class AWSLoggingTest(unittest.TestCase):
```
Comment (tomchop) [inline tests/lib/collectors/aws_logging.py:12]: _Ideally_, we'd also test if `log_client.LookupEvents` is called as expected (e.g. with the properly formatted times) and that the containers are properly added.

**Review by tomchop — CHANGES_REQUESTED** [review]
Some minor nits. Thanks a lot!

File: dftimewolf/lib/containers/containers.py (hunk 1 above)
Comment (jonathan-greig) [reply dftimewolf/lib/containers/containers.py:115]: Realised the docstring was lying and I was actually passing/storing datetime objects. __init__ now casts to str and I've updated the docstrings.

File: dftimewolf/lib/preflights/cloud_token.py (hunk 3 above)
Comment (jonathan-greig) [reply dftimewolf/lib/preflights/cloud_token.py:55]: Added more detail to the method docstring.

File: dftimewolf/lib/preflights/cloud_token.py (hunk 4 above)
Comment (jonathan-greig) [reply dftimewolf/lib/preflights/cloud_token.py:50]: The default profile (as resolved by boto3) - have clarified in docstring.

**Comment by tomchop** [comment]
BTW, just changed the main branch name from `master` to `main`, so you might wanna check out these instructions to adjust your fork: https://docs.github.com/en/github/administering-a-repository/renaming-a-branch#updating-a-local-clone-after-a-branch-name-changes

File: tests/lib/collectors/aws_logging.py [hunk 6]
```diff
@@ -3,21 +3,85 @@
...
+    start_time = '2021-01-01 00:00:00'
+    end_time = '2021-01-02 00:00:00'
+    aws_logging_collector.SetUp(self.FAKE_ZONE, query_filter=query_filter,
+        start_time=start_time, end_time=end_time)
+    aws_logging_collector.Process()
+
+    # pylint: disable=protected-access,no-member
+    aws_logging_collector._log_client.LookupEvents.assert_called_with(
+        qfilter=query_filter, starttime=dt.fromisoformat(start_time),
+        endtime = dt.fromisoformat(end_time))
```
Comment (tomchop) [inline tests/lib/collectors/aws_logging.py:69]: In tests, it's best to refer to literal values whenever possible (e.g. replacing `start_time` by `'2021-01-01 00:00:00'`

Comment (tomchop) [inline tests/lib/collectors/aws_logging.py:69]:
```suggestion
    aws_logging_collector._log_client.LookupEvents.assert_called_with(
        qfilter=query_filter, starttime=dt.fromisoformat(start_time),
        endtime=dt.fromisoformat(end_time))
```

**Review by tomchop — CHANGES_REQUESTED** [review]
A few nits, otherwise LGTM

**Review by tomchop — APPROVED** [review]

**Pull request merged**